bundles/alpha.model/src/alpha/model/util/Facet.xtend

```diff
@@ -77,18 +86,82 @@ class Facet {
 	
 	/** Extract the information from the given set. */
 	new(ISLBasicSet basicSet, ArrayList<Integer> saturatedInequalityIndices) {
-		equalities = DomainOperations.toISLEqualityMatrix(basicSet)
-		indexCount = basicSet.dim(ISLDimType.isl_dim_set)
-		indexInequalities = getInequalities(basicSet, indexCount, true)
+		val basicSetNoRedundancies = basicSet.copy.removeRedundancies
+		
+		space = basicSetNoRedundancies.space 
+		equalities = DomainOperations.toISLEqualityMatrix(basicSetNoRedundancies)
+		indexCount = basicSetNoRedundancies.dim(ISLDimType.isl_dim_set)
+		
+		val allInequalities = getInequalities(basicSetNoRedundancies, indexCount, true)
+		val inequalities = separateEffectivelySaturatedInequalities(allInequalities, space)
+		indexInequalities = inequalities.key
+		effectivelySaturatedInequalities = inequalities.value
+		
+		if (effectivelySaturatedInequalities.nbRows % 2 != 0) {
+			throw new Exception('Failed to create lattice, there should be an even number of (or zero) effectively saturated constraints')
+		}
+		
 		indexInequalityCount = indexInequalities.nbRows
-		isBounded = basicSet.bounded
-		isEmpty = basicSet.empty
+		isBounded = basicSetNoRedundancies.bounded
+		isEmpty = basicSetNoRedundancies.empty
 		parameterEqualityCount = countParameterConstraints(equalities, indexCount)
-		parameterInequalities = getInequalities(basicSet, indexCount, false)
+		parameterInequalities = getInequalities(basicSetNoRedundancies, indexCount, false)
 		this.saturatedInequalityIndices = saturatedInequalityIndices
-		space = basicSet.space 
+		
+		dimensionality = dimensionality(effectivelySaturatedInequalities, equalities, indexCount)
+	}
+	
+	/** Separates the effectively saturated inequalities from true unsaturated inequalities */
+	def static separateEffectivelySaturatedInequalities(ISLMatrix allInequalities, ISLSpace space) {
```

Comment:
The `separateEffectivelySaturatedInequalities` and `buildFromLongMatrixEvenIfEmpty` are only used inside of construction. Can we make them `protected` and move them to the "Private Methods" section?